distro/run/core/src/main/java/org/camunda/bpm/run/CamundaBpmRunConfiguration.java

```diff
@@ -16,64 +16,57 @@
-import org.camunda.bpm.spring.boot.starter.configuration.CamundaDeploymentConfiguration;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.util.List;
+
 @EnableConfigurationProperties(CamundaBpmRunProperties.class)
 @Configuration
 @AutoConfigureAfter({ CamundaBpmAutoConfiguration.class })
 public class CamundaBpmRunConfiguration {
 
-  @Autowired
-  CamundaBpmRunProperties camundaBpmRunProperties;
-
   @Bean
   @ConditionalOnProperty(name = "enabled", havingValue = "true", prefix = CamundaBpmRunLdapProperties.PREFIX)
-  public LdapIdentityProviderPlugin ldapIdentityProviderPlugin() {
-    return camundaBpmRunProperties.getLdap();
+  public LdapIdentityProviderPlugin ldapIdentityProviderPlugin(CamundaBpmRunProperties properties) {
+    return properties.getLdap();
   }
 
   @Bean
   @ConditionalOnProperty(name = "enabled", havingValue = "true", prefix = CamundaBpmRunAdministratorAuthorizationProperties.PREFIX)
-  public AdministratorAuthorizationPlugin administratorAuthorizationPlugin() {
-    return camundaBpmRunProperties.getAdminAuth();
+  public AdministratorAuthorizationPlugin administratorAuthorizationPlugin(CamundaBpmRunProperties properties) {
+    return properties.getAdminAuth();
   }
 
   @Bean
-  public ProcessEngineConfigurationImpl processEngineConfigurationImpl(List<ProcessEnginePlugin> processEnginePlugins) {
-    final SpringProcessEngineConfiguration configuration = new CamundaBpmRunProcessEngineConfiguration();
-
-    // register process engine plugins defined in yaml
-    List<CamundaBpmRunProcessEnginePluginProperty> yamlPluginsInfo = camundaBpmRunProperties.getProcessEnginePlugins();
-    CamundaBpmRunProcessEnginePluginHelper.registerYamlPlugins(processEnginePlugins, yamlPluginsInfo);
-
-    configuration.getProcessEnginePlugins().add(new CompositeProcessEnginePlugin(processEnginePlugins));
-    return configuration;
+  public ProcessEngineConfigurationImpl processEngineConfigurationImpl(
+      List<ProcessEnginePlugin> processEnginePlugins,
+      CamundaBpmRunProperties properties,
+      CamundaBpmRunDeploymentConfiguration deploymentConfig,
+      @Value("${camunda.bpm.run.deploy-changed-only:true}") boolean deployChangedOnly)
+  {
+    var normalizedDeploymentDir = deploymentConfig.getNormalizedDeploymentDir();
+    return new CamundaBpmRunProcessEngineConfiguration(properties, normalizedDeploymentDir, deployChangedOnly,
+        processEnginePlugins);
   }
 
   @Bean
-  public static CamundaDeploymentConfiguration camundaDeploymentConfiguration() {
```

Comment:
The `static` identifier shouldn't exist in a `@Configuration` class.

Reply:
I found [this](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html#:~:text=Special%20consideration%20must,to%20BeanFactoryPostProcessor.) in JavaDoc.

It mentions that certain beans which have a requirement to not be part of the Spring Bean Lifecycle (e.g perform some initialisation) can avoid it by declaring the method with static.

I'm sceptical if that is the case with `CamundaDeploymentConfiguration` or any other config class. Never seen that need in any spring app so far :)

Reply:
If we are we unsure we could confirm it with the developer who wrote it.